hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -667,6 +681,12 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
             if (now > timeToStartJoin) {
                 startJoin();
             }
+            if (node.isMaster()) {
+                long startTime = clusterClock.getClusterStartTime();
+                nodeEngine.getOperationService().invokeOnTarget(SERVICE_NAME,
```

Comment:
Shouldn't this be part of join process? Instead of sending a separate operation...